src/path_manager.c

```diff
@@ -1136,7 +1136,8 @@ struct mptcpd_pm *mptcpd_pm_create(struct mptcpd_config const *config)
          *      @c mptcpd_plugin_unload() is called.
          */
         if (!mptcpd_plugin_load(config->plugin_dir,
-                                config->default_plugin)) {
+                                config->default_plugin,
+                                pm)) {
```

Comment:
The introduction of this argument was intended to allow plugins to perform `mptcpd_pm` related operations, i.e. those in `<mptcpd/path_manager.h>`,  early on in their `init` function.  However, those operations will fail if mptcpd is not fully initialized.  Specifically, the generic netlink family for the kernel's netlink path manager will not have "appeared" prior to plugin load.  This call to `mptcpd_plugin_load()` should be moved to the end of the `family_appeared()` callback function in this file.  Doing so would also help ensure that mptcpd is always in the "ready" state prior to plugin load.

Assuming this is what we want to do, should we create a separate PR for the patch, or add it to this PR?

Reply:
I'd lean toward another PR